docker/amazonlinux-latest-amd64-openssl-1.0.x/Dockerfile

```diff
@@ -0,0 +1,51 @@
+# syntax=docker/dockerfile:1
+
+# substitute for centos
+FROM amazonlinux:latest
+
+# Node.js 18 isn't compatible with Amazon Linux due to glibc version. It would throw with:
+#   node: /lib64/libm.so.6: version `GLIBC_2.27' not found (required by node)
+#   node: /lib64/libc.so.6: version `GLIBC_2.28' not found (required by node)
+ARG NODE_VERSION="16.6.0"
+ARG ARCH="x64"
+
+WORKDIR /usr/src/app
+COPY --from=app . ./
+COPY --from=utils . ./
+
+# Install system dependencies
+RUN yum update -y \
+  && yum install -y tar xz
+
+# Install Node.js
+RUN curl -fsSLO --compressed "https://nodejs.org/dist/v${NODE_VERSION}/node-v${NODE_VERSION}-linux-${ARCH}.tar.xz" \
+  && tar -xJf "node-v${NODE_VERSION}-linux-${ARCH}.tar.xz" -C /usr/local --strip-components=1 --no-same-owner \
+  && rm "node-v${NODE_VERSION}-linux-${ARCH}.tar.xz" \
+  && npm i -g yarn
+
+# Both OpenSSL 1.0 and OpenSSL 3.0 are expected to be on this system
```

Comment:
Interesting, was not aware of that.
Would that mean a Engine with a OpenSSL dynamic linking would also work there?

Reply:
Thanks for raising that to my attention. First of all, `libssl3.so` is misleading. I inadvertently treated it as `openssl-3.0.x`, but it's not that. Only `openssl-1.0.x` is installed in `amazonlinux` by default.

> Would that mean a Engine with a OpenSSL dynamic linking would also work there?

Apparently, yes. E.g., if we install `libssl-3.0.x` and manually specify we should use an engine like `debian-openssl-3.0.x`, the system libraries shipped with `libssl.so.3` would be found. Currently, it's unclear to me whether that'd work with `libssl-1.1.x` as well.

I've tried locally using `amazonlinux` with a `sqlite` schema using `binaryTargets = ["native", "debian-openssl-1.0.x", "debian-openssl-3.0.x"]`, where `native` is resolved as `rhel-openssl-1.0.x`.

Using `ldd`, we can see the dynamically linked libraries for each query engine. We can see that the `*-openssl-3.0.x` engines try to dynamically load `libssl.so.3` and `libcrypto.so.3`, which isn't installed:

- `ldd libquery_engine-debian-openssl-3.0.x.so.node`

  ```sh
  libz.so.1 => /lib64/libz.so.1 (0x0000004002a87000)
  libssl.so.3 => not found
  libcrypto.so.3 => not found
  libgcc_s.so.1 => /lib64/libgcc_s.so.1 (0x0000004002c9e000)
  librt.so.1 => /lib64/librt.so.1 (0x0000004002eb4000)
  libpthread.so.0 => /lib64/libpthread.so.0 (0x00000040030bc000)
  libm.so.6 => /lib64/libm.so.6 (0x00000040032da000)
  libdl.so.2 => /lib64/libdl.so.2 (0x000000400361c000)
  libc.so.6 => /lib64/libc.so.6 (0x0000004003820000)
  /lib64/ld-linux-x86-64.so.2 (0x0000004000000000)
  ```

- `ldd libquery_engine-debian-openssl-1.0.x.so.node`

  ```sh
  libz.so.1 => /lib64/libz.so.1 (0x0000004002cea000)
  libgcc_s.so.1 => /lib64/libgcc_s.so.1 (0x0000004002f01000)
  librt.so.1 => /lib64/librt.so.1 (0x0000004003117000)
  libpthread.so.0 => /lib64/libpthread.so.0 (0x000000400331f000)
  libm.so.6 => /lib64/libm.so.6 (0x000000400353d000)
  libdl.so.2 => /lib64/libdl.so.2 (0x000000400387d000)
  libc.so.6 => /lib64/libc.so.6 (0x0000004003a81000)
  /lib64/ld-linux-x86-64.so.2 (0x0000004000000000)
  ```

- `ldd libquery_engine-rhel-openssl-3.0.x.so.node`

  ```sh
  libz.so.1 => /lib64/libz.so.1 (0x00000040029c7000)
  libssl.so.3 => not found
  libcrypto.so.3 => not found
  libgcc_s.so.1 => /lib64/libgcc_s.so.1 (0x0000004002bde000)
  librt.so.1 => /lib64/librt.so.1 (0x0000004002df4000)
  libpthread.so.0 => /lib64/libpthread.so.0 (0x0000004002ffc000)
  libm.so.6 => /lib64/libm.so.6 (0x000000400321a000)
  libdl.so.2 => /lib64/libdl.so.2 (0x000000400355c000)
  libc.so.6 => /lib64/libc.so.6 (0x0000004003760000)
  /lib64/ld-linux-x86-64.so.2 (0x0000004000000000)
  ```

- `ldd libquery_engine-rhel-openssl-1.0.x.so.node`

  ```sh
  libz.so.1 => /lib64/libz.so.1 (0x0000004002c27000)
  libgcc_s.so.1 => /lib64/libgcc_s.so.1 (0x0000004002e3e000)
  librt.so.1 => /lib64/librt.so.1 (0x0000004003054000)
  libpthread.so.0 => /lib64/libpthread.so.0 (0x000000400325c000)
  libm.so.6 => /lib64/libm.so.6 (0x000000400347a000)
  libdl.so.2 => /lib64/libdl.so.2 (0x00000040037ba000)
  libc.so.6 => /lib64/libc.so.6 (0x00000040039be000)
  /lib64/ld-linux-x86-64.so.2 (0x0000004000000000)
  ```

If I install `libssl-3.0.x` manually, and run `ldd` again on the `*-openssl-3.0.x` engines, I see that the dynamic link is satisfied:

- `ldd libquery_engine-debian-openssl-3.0.x.so.node`

  ```sh
  libz.so.1 => /lib64/libz.so.1 (0x0000004002a89000)
  libssl.so.3 => /lib64/libssl.so.3 (0x0000004002ca0000)
  libcrypto.so.3 => /lib64/libcrypto.so.3 (0x0000004002f42000)
  libgcc_s.so.1 => /lib64/libgcc_s.so.1 (0x00000040035a9000)
  librt.so.1 => /lib64/librt.so.1 (0x00000040037bf000)
  libpthread.so.0 => /lib64/libpthread.so.0 (0x00000040039c7000)
  libm.so.6 => /lib64/libm.so.6 (0x0000004003be5000)
  libdl.so.2 => /lib64/libdl.so.2 (0x0000004003f27000)
  libc.so.6 => /lib64/libc.so.6 (0x000000400412b000)
  /lib64/ld-linux-x86-64.so.2 (0x0000004000000000)
  ```

- `ldd libquery_engine-rhel-openssl-3.0.x.so.node`

  ```sh
  libz.so.1 => /lib64/libz.so.1 (0x00000040029c9000)
  libssl.so.3 => /lib64/libssl.so.3 (0x0000004002be0000)
  libcrypto.so.3 => /lib64/libcrypto.so.3 (0x0000004002e82000)
  libgcc_s.so.1 => /lib64/libgcc_s.so.1 (0x00000040034e9000)
  librt.so.1 => /lib64/librt.so.1 (0x00000040036ff000)
  libpthread.so.0 => /lib64/libpthread.so.0 (0x0000004003907000)
  libm.so.6 => /lib64/libm.so.6 (0x0000004003b25000)
  libdl.so.2 => /lib64/libdl.so.2 (0x0000004003e67000)
  libc.so.6 => /lib64/libc.so.6 (0x000000400406b000)
  /lib64/ld-linux-x86-64.so.2 (0x0000004000000000)
  ```

Reply:
On Amazon Linux 2022 `openssl-3.0.x` is installed by default, so dynamically linking for `libssl` already applies

Reply:
(I will mark this conversation as resolved to merge the PR. If anything else seems missing to you, feel free to add a new comment after the PR is merged).